packages/shared/charts/src/components/Chart.tsx

```diff
@@ -0,0 +1,155 @@
+import { createElement, Fragment, ReactElement, useCallback, useMemo } from "react";
+import ReactPlotlyChartComponent from "react-plotly.js";
+import { Config, Data, Layout } from "plotly.js";
+import deepmerge from "deepmerge";
+import { Playground, useChartsPlaygroundState } from "./Playground/Playground";
+import { CodeEditor } from "./Playground/CodeEditor";
+import { ifNonEmptyStringElseEmptyObjectString } from "./Playground/utils";
+
+type ChartDataType = Partial<Data> & {
+    customSeriesOptions: string | undefined;
+};
+export interface ChartProps {
+    data: ChartDataType[];
+    configOptions: Partial<Config>;
+    layoutOptions: Partial<Layout>;
+    seriesOptions: Partial<Data>;
+    customConfig: string | undefined;
+    customLayout: string | undefined;
+}
+
+export const Chart = ({
+    data,
+    configOptions,
+    layoutOptions,
+    seriesOptions,
+    customConfig,
+    customLayout
+}: ChartProps): ReactElement => {
+    const forceReactPlotlyUpdate = useCallback(() => {
```

Comment:
Do we really need this trigger?
the documentation says that if we use `{responsive: true}` the chart will resize every time when browser window size changes 
[PlotlyResponsive](https://plotly.com/javascript/responsive-fluid-layout/#responsive-plots)

Reply:
- The problem that I'm trying to solve is that for some reason when we render the plots, they don't match the container properly, but spill outside of it. Most likely this has something to do with how we handle dimensions: set dimensions on outer container and use 100% height and width for this chart.
- LineChart already adds the `responsive` flag, but unfortunately that only works every time the browser window size changes (this `resize` event is received). When the initial chart renders, there is no resize event cuz the browser doesn't resize.
- To address this, I manually dispatched such an event on initialization of the chart. 
- But that turned out not to work 100%, because the data can still be loading while the chart is initialized. A fix is in 13f7d46